frontend/packages/kubevirt-plugin/src/components/vms/vm.tsx

```diff
@@ -62,42 +91,103 @@ const VMHeader = () =>
-  const uid = getUID(vm);
-  const lookupID = getBasicID(vm);
-
+  const name = getName(obj);
+  const namespace = getNamespace(obj);
+  const uid = getUID(obj);
+  const lookupID = getBasicID(obj);
   const migration = migrationLookup[lookupID];
-  const vmi = vmiLookup[lookupID];
-  const vmStatus = getVMStatus({ vm, vmi, pods, migrations });
+
+  let vm = null;
+  let vmi = null;
+  let status = null;
+  let vmStatus = null;
+  let actions = [Kebab.factory.ModifyLabels, Kebab.factory.ModifyAnnotations, Kebab.factory.Delete];
+
+  if (obj.kind === VirtualMachineModel.kind) {
+    vm = obj as VMKind;
+    vmi = vmiLookup[lookupID];
+    vmStatus = getVMStatus({ vm: vm as VMKind, vmi, pods, migrations });
+    status = <VMStatus vm={vm} vmi={vmi} pods={pods} migrations={migrations} />;
+    actions = menuActions;
+  } else {
+    vmi = obj as VMIKind;
+    status = <Status status={getPhase(vmi)} />;
+  }
 
   return (
     <TableRow id={uid} index={index} trKey={key} style={style}>
       <TableData className={dimensify()}>
-        <ResourceLink kind={VirtualMachineModel.kind} name={name} namespace={namespace} />
+        {vm ? (
+          <ResourceLink kind={VirtualMachineModel.kind} name={name} namespace={namespace} />
+        ) : (
+          <ResourceLinkPopover
+            kind={VirtualMachineInstanceModel.kind}
+            name={name}
+            namespace={namespace}
+            isDisabled
+            message="No VM"
+            linkMessage="VMI Dedatails page"
+          >
+            <div>
+              This VMI doesn’t have an owner VM since it might have been created outside of the
+              console.
+            </div>
+          </ResourceLinkPopover>
+        )}
+      </TableData>
+      <TableData className={dimensify()}>
+        {vmi ? (
+          <ResourceLink
+            kind={VirtualMachineInstanceModel.kind}
+            name={getName(vmi)}
+            namespace={namespace}
+          />
+        ) : (
+          <ResourceLinkPopover
+            kind={VirtualMachineModel.kind}
+            name={name}
+            namespace={namespace}
+            isDisabled
+            message="No Instance"
+            linkMessage="VM Dedatails page"
+          >
+            <div>
+              This VMI is currently off.
+              <br />
+              For further details please click its owner VM link below.
+            </div>
+          </ResourceLinkPopover>
+        )}
       </TableData>
       <TableData className={dimensify()}>
         <ResourceLink kind={NamespaceModel.kind} name={namespace} title={namespace} />
       </TableData>
+      <TableData className={dimensify()}>{status}</TableData>
+      <TableData className={dimensify()}>
+        {vmi && fromNow(vmi.metadata.creationTimestamp)}
+      </TableData>
       <TableData className={dimensify()}>
-        <VMStatus vm={vm} vmi={vmi} pods={pods} migrations={migrations} />
+        {vmi && vmi.status.nodeName && (
```

Comment:
Lets try Typescript 3.7!

`vmi?.status?.nodeName`

Reply:
Nice ! ... but the linter doesn't like it ... moved the `vmi.status.nodeName` to a selector.